tests/alldb_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// alldb_test.go contains end to end tests that require all environment variables to be defined.
+package tests
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+// requireAllVars skips the given test if at least one environment variable is undefined.
+func requireAllVars(t *testing.T) {
+	var allVars []string
+	allVars = append(allVars, *mysqlConnName, *mysqlUser, *mysqlPass, *mysqlDb)
+	allVars = append(allVars, *postgresConnName, *postgresUser, *postgresPass, *postgresDb)
+	allVars = append(allVars, *sqlserverConnName, *sqlserverUser, *sqlserverPass, *sqlserverDb)
+
+	for _, envVar := range allVars {
+		if envVar == "" {
+			t.Skip("skipping test, all environment variable must be defined")
+		}
+	}
+}
+
+// Test to verify that when a proxy client serves multiple instances that can all be successfully dialed,
+// the health check readiness endpoint serves http.StatusOK.
+func TestMultiInstanceDial(t *testing.T) {
+	requireAllVars(t)
+
+	binPath, err := compileProxy()
```

Comment:
There's a helper object for starting/running the proxy as a binary, so you don't need to compile it each time.

Take a look at at this example: https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/cfd5259c626154b4ad689c85370d775938aa5d78/tests/connection_test.go#L66-L75